tofu/spectro/_rockingcurve.py

```diff
@@ -0,0 +1,345 @@
+    psi0_dre = -re*(lamb**2)*(
+        6.*(Zo + dfo_re) + 3.*(Zsi + dfsi_re)
+        )/(np.pi*V)   # psiop
+    # zero-order imaginary part (averaged)
+    psi0_im = -re*(lamb**2)*(6.*fo_im + 3.*fsi_im)/(np.pi*V)    # psios
+
+    # Integrated reflectivity for crystals models: perfect (Darwin model) &
+    # ideally mosaic thick crystal
+    # -------------------------------------------------------------
+
+    R_per = Zo*F_re*re*lamb**2*(1. + abs(np.cos(2.*theta)))/(
+        6.*np.pi*V*np.sin(2.*theta)
+    )
+    R_mos = F_re**2*re**2*lamb**3*(1. + (np.cos(2.*theta))**2)/(
+        4.*mu*V**2*np.sin(2.*theta)
+    )
+
+    # Rocking curve and integrated reflectivity with "dynamical" model vs angle
+    # and for the 2 states of photon polarization
+    # -------------------------------------------------------------------------
+
+    # incident wave polarization (normal & parallel components)
+    polar = np.r_[1., abs(np.cos(2.*theta))]
+    # variables of simplification y, dy, g, L
+    g = psi0_im/(polar*psi_re)
+    y = np.linspace(-10., 10., 501)    # ay
+    dy = np.zeros(501) + 0.1    # day
+    al = np.full((2, 501), 0.)
+
+    power_ratio = np.full((al.shape), np.nan)    # phpo
+    th = np.full((al.shape), np.nan)    # phpo
+    rr = np.full((polar.shape), np.nan)
+    for i in range(al[:, 0].size):
+        al[i, ...] = (y**2 + g[i]**2 + np.sqrt(
+            (y**2 - g[i]**2 + kk**2 - 1.)**2 + 4.*(g[i]*y - rek)**2
+            ))/np.sqrt((kk**2 - 1.)**2 + 4.*rek**2)
+        # reflecting power P_DIN
+        power_ratio[i, ...] = (Fmod/Fbmod)*(
+            al[i, :] - np.sqrt((al[i, :]**2 - 1.))
+        )
+        power_ratiob = power_ratio[i, ...]
+        # intensity scale on the glancing angle (y=kk/g)
+        th[i, ...] = (y*polar[i]*psi_re - psi0_dre)/np.sin(2.*theta)
+        # integration of the power ratio over dy
+        rhy = np.sum(dy*power_ratiob)
+        # diffraction radiation
+        # r(i=0): normal component & r(i=1): parallel component
+        rr[i, ...] = (polar[i]*psi_re*rhy)/np.sin(2.*theta)
+    R_dyn = np.sum(rr)/2.
+    if R_dyn < 1e-7:
+        msg = (
+            "Please check the equations for integrated reflectivity:\n"
+            "the value of R_din ({}) is less than 1e-7.\n".format(R_dyn)
+        )
+        raise Exception(msg)
+
+    fmax = np.max(power_ratio[0] + power_ratio[1])
+    det = (2.*R_dyn)/fmax
+
+    # Plot power ratio
+    # ----------------
+
+    fig1 = plt.figure(figsize=(8, 6))
+    gs = gridspec.GridSpec(1, 1)
+    ax = fig1.add_subplot(gs[0, 0])
+    ax.set_title(
+        'Qz, ' + f'({ih},{ik},{il})' + fr', $\lambda$={lamb} A' +
+        fr', Bragg angle={np.round(theta, decimals=3)} rad'
+    )
+    ax.set_xlabel(r'$\theta$-$\theta_{B}$ (rad)')
+    ax.set_ylabel('P$_H$/P$_0$')
+    ax.plot(th[0, :], power_ratio[0, :], label='normal component')
+    ax.plot(th[1, :], power_ratio[1, :], label='parallel component')
+    ax.legend()
```

Comment:
Same comment: introduce a `plot_power_ratio=None` keyword argument to the routine and move this part to a dedicated sub-routine called only if True